Issue: 
 	- ABE block copy kernel panic with Power-On off tests.

ChangeLog:

Date:2011July12
	- if there was any work waiting then we run it now and
		wait for it's completion.
	- Flush the workqueue platform_driver's shutdown(). 

Date:2011July09
	- ABE_CLKSTCTRL = 3 		<-- HW AUTO
	- ABE_L4ABE_CLKCTRL = 1 	<-- HW AUTO
	
	Ensure the below modules are explicitly enabled 
	before accessing,

	abe_dsp_pm_get ->
		- ABE_AESS_CLKCTRL = 2 		<-- Module explicitly enabled

	pm_runtime_get_sync mcpdm->dev ->
		- ABE_PDM_CLKCTRL = 2		<-- Module explicitly enabled


Index: kernel/sound/soc/omap/omap-mcpdm.c
===================================================================
--- kernel.orig/sound/soc/omap/omap-mcpdm.c	2011-07-12 22:53:33.861998353 +0530
+++ kernel/sound/soc/omap/omap-mcpdm.c	2011-07-12 22:54:17.531998785 +0530
@@ -993,6 +993,26 @@
 	return 0;
 }
 
+static void asoc_mcpdm_shutdown(struct platform_device *pdev)
+{
+	struct omap_mcpdm *mcpdm = platform_get_drvdata(pdev);
+	int ret = 0;
+
+	/* cancel any work waiting to be queued. */
+	ret = cancel_delayed_work_sync(&mcpdm->delayed_abe_work);
+
+	/* 
+	 * if there was any work waiting then we run it now and
+	 * wait for it's completion 
+	 */
+	if (ret) 
+	{
+		schedule_delayed_work(&mcpdm->delayed_abe_work, 0);
+		flush_scheduled_work();
+	}
+
+}
+
 static struct platform_driver asoc_mcpdm_driver = {
 	.driver = {
 			.name = "omap-mcpdm-dai",
@@ -1001,6 +1021,7 @@
 
 	.probe = asoc_mcpdm_probe,
 	.remove = __devexit_p(asoc_mcpdm_remove),
+	.shutdown = asoc_mcpdm_shutdown,
 };
 
 static int __init snd_omap_mcpdm_init(void)
